### PR TITLE
FEAT: Bayside VIC scraper

### LINKED ISSUE #70: [New Scraper] VIC: Bayside
**Council Name:**
Bayside Council - VIC

**Link to agenda webpage:**
https://www.bayside.vic.gov.au/council/council-meetings-agendas-and-minutes

**Webpage quirks**
Is there anything that might make scraping difficult?
Looks like a relatively straightforward set of links to the agenda.

**Additional context**
Add any other context or screenshots about the scraper request here.
image
image

### PR DESCRIPTION
**Pull request by ryanward-io**
Closes #70.

If we also scrape the PDF I can probably get the time too but it seemed out of scope.

Also in the `ScraperReturn` class I'm not sure the value of scraping `name`. The comment says `# council name` but is this supposed to be the name of the agenda? Or is it council name to verify that we have the right council when scraping?

### CODE REVIEW

**Comment by ryanward-io** [comment]
@ArchangelEkim Like general meeting, planning meeting, special meeting, annual meeting, etc?

**Comment by ArchangelEkim** [comment]
yeah, more or less.

File: council_scrapers/scrapers/vic/bayside.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            {scraper_return.webpage_url}
+            {scraper_return.download_url}"""
+        )
+
+        self.logger.info(f"{self.council_name} scraper finished successfully")
+        return scraper_return
+
+
+if __name__ == "__main__":
+    scraper = BaysideScraper()
```
Comment (ArchangelEkim) [inline council_scrapers/scrapers/vic/bayside.py:48]:
```suggestion
    scraper = BaysideVicScraper()
```
Same here

File: council_scrapers/scrapers/vic/bayside.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        raw_html = self.fetcher.fetch_with_requests(initial_webpage_url)
+
+        # Find latest agenda link
+        soup = BeautifulSoup(raw_html, "html.parser")
+        agenda_list = soup.find("div", class_="page__body")
+        latest_agenda = agenda_list.find("a")
+
+        # Scrape the data
+        raw_date = re.search(self.DATE_REGEX, latest_agenda.text).group()
```
Comment (ArchangelEkim) [inline council_scrapers/scrapers/vic/bayside.py:27]:
```suggestion
        raw_date = re.search(self.date_regex, latest_agenda.text).group()
```

File: council_scrapers/scrapers/vic/bayside.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+        # Find latest agenda link
+        soup = BeautifulSoup(raw_html, "html.parser")
+        agenda_list = soup.find("div", class_="page__body")
+        latest_agenda = agenda_list.find("a")
+
+        # Scrape the data
+        raw_date = re.search(self.date_regex, latest_agenda.text).group()
+        date = "-".join([datetime.strptime(raw_date, "%d %B %Y").strftime("%Y-%m-%d")])
+        time = None  # 6:30PM from past few agendas but only on PDF
```
Comment (ArchangelEkim) [inline council_scrapers/scrapers/vic/bayside.py:29]:
```suggestion
        time = "18:30"  # 6:30PM from past few agendas but only on PDF
```

6:30pm is explicitly stated on the Bayside council website as the beginning time for a meeting.

**Review by ArchangelEkim — APPROVED** [review]

**Comment by ryanward-io** [comment]
The failure is a false positive because the test only compare the result to when main was last run. I think it means that you can never make commits from Github GUI and always have to make commits locally and manually delete and update the test data with how the code currently works. I'll redo locally now and push up

**Comment by ArchangelEkim** [comment]
Okay, thats certainly a bug to possibly look at!

**Comment by ryanward-io** [comment]
Tests passing now - feel free to merge

**Pull request merged**